clients/gtest/dot_ex_gtest.cpp

```diff
@@ -0,0 +1,265 @@
+// incx , incy must > 0, otherwise there is no real computation taking place,
+// but throw a message, which will still be detected by gtest
+const vector<vector<int>> incx_incy_range = {
+    {1, 1},
+    {-1, -1},
+};
+
+const double stride_scale_range[] = {1.0, 2.5};
+
+const int batch_count_range[] = {-1, 0, 1, 2, 10};
+
+const vector<vector<hipblasDatatype_t>> precisions{
+    // Not supported in cuBLAS
+    {HIPBLAS_R_16B, HIPBLAS_R_16B, HIPBLAS_R_16B, HIPBLAS_R_32F},
+    {HIPBLAS_R_16F, HIPBLAS_R_16F, HIPBLAS_R_16F, HIPBLAS_R_16F},
+
+    // Supported in both rocBLAS and cuBLAS
+    {HIPBLAS_R_16F, HIPBLAS_R_16F, HIPBLAS_R_16F, HIPBLAS_R_32F},
+    {HIPBLAS_R_32F, HIPBLAS_R_32F, HIPBLAS_R_32F, HIPBLAS_R_32F},
+    {HIPBLAS_R_64F, HIPBLAS_R_64F, HIPBLAS_R_64F, HIPBLAS_R_64F},
+    {HIPBLAS_C_32F, HIPBLAS_C_32F, HIPBLAS_C_32F, HIPBLAS_C_32F},
+    {HIPBLAS_C_64F, HIPBLAS_C_64F, HIPBLAS_C_64F, HIPBLAS_C_64F}};
+
+const bool is_fortran[] = {false, true};
+
+/* ===============Google Unit Test==================================================== */
+
+class dot_ex_gtest : public ::TestWithParam<dot_ex_tuple>
+{
+protected:
+    dot_ex_gtest() {}
+    virtual ~dot_ex_gtest() {}
+    virtual void SetUp() {}
+    virtual void TearDown() {}
+};
+
+Arguments setup_dot_ex_arguments(dot_ex_tuple tup)
+{
+    Arguments arg;
+    arg.N                                     = std::get<0>(tup);
+    int         N                             = std::get<0>(tup);
+    vector<int> incx_incy                     = std::get<1>(tup);
+    arg.stride_scale                          = std::get<2>(tup);
+    arg.batch_count                           = std::get<3>(tup);
+    vector<hipblasDatatype_t> precision_types = std::get<4>(tup);
+    arg.fortran                               = std::get<5>(tup);
+
+    arg.incx = incx_incy[0];
+    arg.incy = incx_incy[1];
+
+    arg.a_type       = precision_types[0];
+    arg.b_type       = precision_types[1];
+    arg.c_type       = precision_types[2];
+    arg.compute_type = precision_types[3];
+
+    arg.timing
+        = 0; // disable timing data print out. Not supposed to collect performance data in gtest
+
+    return arg;
+}
+
+// dot tests
+TEST_P(dot_ex_gtest, dot_ex)
+{
+    // GetParam return a tuple. Tee setup routine unpack the tuple
+    // and initializes arg(Arguments) which will be passed to testing routine
+    // The Arguments data struture have physical meaning associated.
+    // while the tuple is non-intuitive.
+    Arguments       arg    = setup_dot_ex_arguments(GetParam());
+    hipblasStatus_t status = testing_dot_ex(arg);
+    // if not success, then the input argument is problematic, so detect the error message
+    if(status != HIPBLAS_STATUS_SUCCESS)
+    {
+        if(arg.N < 0 || arg.incx < 0 || arg.incy < 0)
+        {
+            EXPECT_EQ(HIPBLAS_STATUS_INVALID_VALUE, status);
+        }
+        else
+        {
+            EXPECT_EQ(HIPBLAS_STATUS_NOT_SUPPORTED, status); // for cuda
```

Comment:
Can we log these unsupported tests to keep track of them?

Reply:
Hm, similar to the  skipping tests aspect done in rocBLAS, you mean?

Reply:
Just so we can count how many not supported by grepping the log, something output.  Not really skipped